src/remote_publish.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2025 qBraid
+#
+# This file is part of the qBraid/publish-environment-action
+#
+# The qBraid/publish-environment-action is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for the qBraid/publish-environment-action, as per Section 15 of the GPL v3.
+
+"""
+Script to create and request to publish a new environment on qBraid.
+
+Requires the following environment variables:
+- ENV_CONFIG_PATH: Path to the environment configuration file.
+- PERSIST_ENV: Boolean value indicating whether to persist the environment.
+- GITHUB_ENV: Path to the GitHub environment file where the environment slug will be written.
+
+"""
+
+import os
+import sys
+
+from qbraid_core.services.environments.client import EnvironmentManagerClient
+from qbraid_core.services.environments.schema import EnvironmentConfig
+
+
+def load_environment_config(config_path: str) -> EnvironmentConfig:
+    """Loads environment configuration from YAML file."""
+    if not os.path.exists(config_path):
+        print(f"Environment configuration file not found at {config_path}")
+        sys.exit(1)
+
+    print(f"Loading environment configuration from {config_path}...")
+    try:
+        config = EnvironmentConfig.from_yaml(config_path)
+        print("Successfully validated environment configuration data.")
+        return config
+    except Exception as e:
+        print(f"Failed to load environment configuration: \n\t{e}")
+        sys.exit(1)
+
+
+def publish_environment(config: EnvironmentConfig, persist_env: bool) -> str:
+    """Publishes the environment and retrieves the environment slug."""
+    try:
+        client = EnvironmentManagerClient()
+        response = client.remote_publish_environment(config=config, persist_env=persist_env)
```

Comment:
Publishing through config had [a bug in the API](https://github.com/qBraid/api/pull/1037). Apparently the config wasn't getting serialized correctly, so unless the change is merged and published will have to hold this.

Reply:
I think it's cleaner if the validation is performed within `qbraid-core` before the request goes out to the API